### PR TITLE
Vfs

### LINKED ISSUE #180: Profile pictures

### LINKED ISSUE #190: Passwords for newly added users
We need to either add it when uploading students, or we need to figure out a way to create them dynamically

### LINKED ISSUE #197: students can access /accountform, and able to create students

### PR DESCRIPTION
**Pull request by ahmedqarmout2**
fixes #197 
fixes #190 
fixes #180

### CODE REVIEW

File: public/js/leaderboard.js [hunk 1]
```diff
@@ -99,13 +99,14 @@ $('#option-attempt').click(function(evt) {
...
     sortLeaderBoard(type.name);
+    setPodiumImages();    
     displayLeaderboard(studentLeaderList);
 }
 
 var setPodiumImages = function() {
-    $('#first').attr('src','img/logo.png');
-    $('#second').attr('src','img/logo.png');
-    $('#third').attr('src','img/logo.png');
+    var date = new Date();
```
Comment (esteeroramy) [inline public/js/leaderboard.js:108]: String interpolation
``` JavaScript
$('#first').attr('src',`/profilePicture/ ${studentLeaderList[0].picture}?${date}`);

File: server.js [hunk 2]
```diff
@@ -809,11 +816,89 @@ app.post('/profilemod', function(req, res) {
...
+/**
+ * fetch the profile picture
+ */
+app.get('/profilePicture/:pictureId', function (req, res) {
+    if (!req.session.user) {
+        return res.redirect('/');
+    }
+
+    var pictureId = req.params.pictureId;
+    if (pictureId === 'null') {
```
Comment (esteeroramy) [inline server.js:897]: Why is `null` here in quotes

File: server.js [hunk 3]
```diff
@@ -809,11 +816,89 @@ app.post('/profilemod', function(req, res) {
...
+        }
+
+        if (fileObj.permission !== common.vfsPermission.PUBLIC) {
+            logger.error(common.formatString('Permission denied to access: {0}',[pictureId]));
+            return res.status(403).send(common.getError(9006));
+        }
+
+        return res.sendFile(fileObj.path, function (err) {
+            if (err) {
+                logger.error(err);
```
Comment (esteeroramy) [inline server.js:918]: Handle this error

File: server/errors.js [hunk 4]
```diff
@@ -107,5 +108,14 @@ const errors = Object.freeze({
     8000: 'failed to add feedback',
     8001: 'failed to get feedback',
     8002: 'failed to remove all feedback',
+
+    //9000 virtual file system
+    9000: 'failed to clean the virtual file system',
```
Comment (esteeroramy) [inline server/errors.js:113]: You need to add these error codes to the UI common.js

**Review by Mohamed-A-Mohamed — APPROVED** [review]
Nice work

**Review by mehtadarshan1 — APPROVED** [review]
great work

File: server.js (hunk 3 above)
Comment (ahmedqarmout2) [reply server.js:918]: this is a different kind of request. There is no handling in the UI for this kind, unless if we want to make a custom handler

File: server/errors.js (hunk 4 above)
Comment (ahmedqarmout2) [reply server/errors.js:113]: fixed

File: server.js (hunk 2 above)
Comment (ahmedqarmout2) [reply server.js:897]: we talked about this offline

File: server.js (hunk 3 above)
Comment (esteeroramy) [reply server.js:918]: Nope, all good

**Review by esteeroramy — APPROVED** [review]

**Comment by ahmedqarmout2** [comment]
virtual file system, and profile pictures

**Pull request merged**